pkgs/tools/networking/ppp/default.nix

```diff
@@ -48,6 +51,8 @@ stdenv.mkDerivation rec {
     "CC=${stdenv.cc.targetPrefix}cc"
   ];
 
+  NIX_LDFLAGS = "-lcrypt";
```

Comment:
Without this the build would fail to link. Not sure if this is the proper way to workaround this.

```
ppp> cc -O2 -g -pipe -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP -pipe -I../include '-DDESTDIR="/nix/store/l0gbyq4xxzc053bswl9slcp7j0hnc3d6-ppp-2.4.9"' -DCHAPMS=1 -DMPPE=1 -DUSE_EAPTLS=1 -DHAS_SHADOW -I/nix/store/0yy468mw8yijm5p7y3gjk8wyf7qzh4g4-openssl-3.0.5-dev/include/openssl -DHAVE_MULTILINK -DUSE_TDB=1 -DPLUGIN -DPPP_FILTER -DINET6=1 -DMAXOCTETS  -Wl,-E -o pppd main.o magic.o fsm.o lcp.o ipcp.o upap.o chap-new.o md5.o ccp.o ecp.o auth.o options.o demand.o utils.o sys-linux.o ipxcp.o tty.o eap.o chap-md5.o session.o md4.o chap_ms.o sha1.o eap-tls.o pppcrypt.o multilink.o tdb.o spinlock.o ipv6cp.o eui64.o -lrt -lssl -lcrypto -ldl -lpcap
ppp> /nix/store/c5d4jq7mjsh8bkhy67k8mxsybj5r7p3x-binutils-2.38/bin/ld: auth.o: in function `check_passwd':
ppp> /build/source/pppd/auth.c:1515: undefined reference to `crypt'
ppp> /nix/store/c5d4jq7mjsh8bkhy67k8mxsybj5r7p3x-binutils-2.38/bin/ld: session.o: in function `session_start':
ppp> /build/source/pppd/session.c:346: undefined reference to `crypt'
ppp> collect2: error: ld returned 1 exit status
```

Reply:
I've noticed that lots of build configuration scripts detect libcrypt very sloppily or not at all. I've also needed to add NIX_* at some places but sometimes a patch will also do. This was the weirdest one: https://github.com/NixOS/nixpkgs/pull/181764/commits/096f0e1d73c80fbc878c99d16a5445d6c95b1252